### PR TITLE
feat: add gov v3 helpers

### PR DESCRIPTION
**Pull request by sendra**
Things to note:
- for aave-proposals etc the "execute as ecosystem reserve" path (which is used for dry run & emit only path) does not work any more. The issue is the ecosystem reserve not having enough eth to pay the fee.

### CODE REVIEW

File: src/GovV3Helpers.sol [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    }
+    return
+      IGovernanceCore(AaveV3EthereumGovV3.GOVERNANCE).createProposal(
+        payloads,
+        votingPortal,
+        ipfsHash
+      );
+  }
+
+  function executePayload(Vm vm, uint40 payloadId) internal {
```
Comment (sakulstra) [inline src/GovV3Helpers.sol]: Should probably have an overload like:
`  function executePayload(Vm vm, address payloadAddress) internal {`
creating the payload first

File: src/GovV3Helpers.sol [hunk 2]
```diff
@@ -0,0 +1,759 @@
...
+    vm.selectFork(prevFork);
+    return (payload.maximumAccessLevelRequired, payloadId);
+  }
+
+  function _findPayloadId(
+    IPayloadsControllerCore payloadsController,
+    IPayloadsControllerCore.ExecutionAction[] memory actions
+  ) private view returns (uint40, IPayloadsControllerCore.Payload memory) {
+    uint40 count = payloadsController.getPayloadsCount();
+    for (uint40 payloadId = count - 1; payloadId >= 0; payloadId++) {
```
Comment (sendra) [inline src/GovV3Helpers.sol]: should be payloadId--

Comment (sakulstra) [reply src/GovV3Helpers.sol]:
```suggestion
    for (uint40 payloadId = count - 1; payloadId >= 0; payloadId--) {
```

File: src/ChainIds.sol [hunk 3]
```diff
@@ -16,3 +17,43 @@ library ChainIds {
...
+    } else if (chainId == ChainIds.OPTIMISM) {
+      newFork = vm.createFork(vm.rpcUrl('optimism'));
+    } else if (chainId == ChainIds.BNB) {
+      newFork = vm.createFork(vm.rpcUrl('bnb'));
+    } else if (chainId == ChainIds.POLYGON) {
+      newFork = vm.createFork(vm.rpcUrl('polygon'));
+    } else if (chainId == ChainIds.FANTOM) {
+      newFork = vm.createFork(vm.rpcUrl('fantom'));
+    } else if (chainId == ChainIds.ZK_SYNC) {
+      newFork = vm.createFork(vm.rpcUrl('tkSync'));
```
Comment (sendra) [inline src/ChainIds.sol]:
```suggestion
      newFork = vm.createFork(vm.rpcUrl('zkSync'));
```

File: src/GovV3Helpers.sol [hunk 4]
```diff
@@ -0,0 +1,904 @@
...
+        withDelegateCall: withDelegateCall,
+        accessLevel: accessLevel,
+        value: value,
+        signature: signature,
+        callData: callData
+      });
+  }
+
+  /**
+   * Registers a payload with the provided actions on the networjḱ PayloadsController
```
Comment (rustboyar) [inline src/GovV3Helpers.sol]: networjk -> network

File: .env.example [hunk 5]
```diff
@@ -11,6 +11,7 @@ RPC_FANTOM=https://rpc.ftm.tools
 RPC_HARMONY=https://api.harmony.one
 RPC_METIS=https://andromeda.metis.io/
 RPC_BASE=https://mainnet.base.org
+RPC_BNB=https://binance.llamarpc.com
```
Comment (kyzia551) [inline .env.example:15]: Should it also include `Sepolia`?
I see it exists in `foundry.toml`

**Review by rustboyar — APPROVED** [review]

File: src/GovV3Helpers.sol [hunk 6]
```diff
@@ -0,0 +1,904 @@
...
+
+  function createPayload(
+    IPayloadsControllerCore.ExecutionAction memory action
+  ) internal returns (uint40) {
+    IPayloadsControllerCore payloadsController = getPayloadsController(block.chainid);
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actions = new IPayloadsControllerCore.ExecutionAction[](1);
+    actions[0] = action;
+    return payloadsController.createPayload(actions);
+  }
```
Comment (kyzia551) [inline src/GovV3Helpers.sol]:
```suggestion
  function createPayload(
    IPayloadsControllerCore.ExecutionAction memory action
  ) internal returns (uint40) {
    IPayloadsControllerCore.ExecutionAction[]
      memory actions = new IPayloadsControllerCore.ExecutionAction[](1);
    actions[0] = action;
    return createPayload(actions);
  }
```

File: src/GovV3Helpers.sol [hunk 7]
```diff
@@ -0,0 +1,904 @@
...
+      actions
+    );
+    return
+      PayloadsControllerUtils.Payload({
+        chain: chainId,
+        accessLevel: accessLevel,
+        payloadsController: address(payloadsController),
+        payloadId: payloadId
+      });
+  }
```
Comment (kyzia551) [inline src/GovV3Helpers.sol]:
```suggestion
  function _buildPayload(
    Vm vm,
    uint256 chainId,
    IPayloadsControllerCore.ExecutionAction memory action
  ) internal returns (PayloadsControllerUtils.Payload memory) {
    IPayloadsControllerCore.ExecutionAction[]
      memory actions = new IPayloadsControllerCore.ExecutionAction[](1);
    actions[0] = action;
    return _buildPayload(vm, chainId, actions);
  }
```

**Review by kyzia551 — CHANGES_REQUESTED** [review]

**Review by kyzia551 — APPROVED** [review]

**Pull request merged**